h2o-core/src/main/java/water/fvec/RollupStats.java

```diff
@@ -431,7 +437,9 @@ public void compute2() {
             rs._tsk.join();
           } else if(rs.isMutating()) // c) => throw IAE
             throw new IllegalArgumentException("Can not compute rollup stats while vec is being modified. (3)");
-        } else { // d) => compute the rollups
+        } else { // rs == null d) => compute the rollups
+          final Vec vec = DKV.getGet(_vecKey);
+          if (vec == null) break; // Vec was deleted since we last checked
```

Comment:
Hmm, I still think it's the same as above, let me elaborate on that and see if I got something wrong:

Why I think it is not needed:

When you remove Vec, you first remove the vec itself, then mark it as mutating (replace any RS with mutating one), then go on to remove the data, then delete the data and in the end, the rollups.

So when someone racily requests rollups while the vec is being deleted, following can happen when his ComputeRollupsTask reaches the home node:
a) the rollups are not there, remove did not happen yet 
     1. he will first acquire the K/V lock (meaning he will succeed inserting Rollups marked as computing)
     2. he will start computing the rollups
         - here either the vec gets deleted in the meantime a you get an NPE (which is ok) or you finish the computation before any data gets deleted
         - proceed to install result 
             now there is three possible states here:
               1. the remove did not reach this rollups, task still has a valid lock
                   this is ok, new rollups are inserted and will later be removed
               2. the remove already marked this vec as mutating - the lock has been broken => throws
               3. the remove is already done, there are no rollups in the store any more => lock has been broken => throws
b) the remove is already in progress, vec is marked as mutating => throws
c) the remove is done, the rollups are no longer there -> it will try to compute the rollups but get an NPE cause the data is no longer there                   

why I think it can lead to infinite loop

The caller loops until he gets valid rollups (from the K/V) or an exception
Now, for the case c) from above:

The remove is done, there is no rollups. He will try to proceed to compute them.
He checks for Vec, does not find it and just returns.
From the callers pont of view, he just launched rollups task which ran successfully, he will attempt DKV.get which will return null
Will continue with his loop.